### PR TITLE
V2 documentation

### PR DESCRIPTION
**Pull request by CIPop**
- User guide
- Platform porting guide and code links
- State machine for connection management
- Rendering PUML state machines for RPC
- Adding missing links (step 2 required by internal link validation system).

### CODE REVIEW

**Review by RLeclair — APPROVED** [review]
LGTM

File: sdk/docs/core/README.md [hunk 1]
```diff
@@ -177,6 +177,138 @@ Note however that cancellation is performed as a best effort; it is not guarante
...
+
+The SDK implements several communication patterns using the [MQTTv5 Protocol](https://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html).
+Application developers can choose between two layers of public APIs:
+
+1. Encoder/decoder (codec) functions that provide adaptation from the MQTT patterns or Azure IoT services and require the application to perform all I/O.
+2. Declarative, stateful functions that rely on a Platform Adaptation Layer.
+
+#### Codec API
+
+The codec API surface provides a set of functions that allows the application developer to take full-control of the MQTT stack, regardless of I/O or threading models. The Codec API surface offers in-memory translation from the MQTT patterns or Azure IoT client data structures to the MQTT protocol (i.e. topics, payloads and properties).
```
[the remark this answers is not in the document]
Comment (CIPop) [reply sdk/docs/core/README.md:190]: Good point: I think that is a missing feature. The codec should provide constants (read `#define`s) with the user property names and potentially type aliases (e.g. `RPC_ENCODING_PROPERTY_TYPE`, etc.)

File: sdk/docs/core/README.md [hunk 2]
```diff
@@ -177,6 +177,138 @@ Note however that cancellation is performed as a best effort; it is not guarante
...
+
+  struct mosquitto* mosq = NULL;       // MQTT handle (in this case for Mosquitto MQTT Client).
+  az_mqtt5_init(&mqtt5, &mosq, NULL);  // associate the az_mqtt5 adapter to a mosquitto MQTT handle.
+                                       // For this to work, the SDK was built with Mosquitto MQTT as 
+                                       // MQTT platform implementation.
+
+  az_mqtt_connection_init( 
+      mqtt_connection, 	      // connection object 
+      &connection_context,    // cancellation context 
+      &mqtt,                  // mqtt abstraction (e.g. over Paho Async)
```
[the remark this answers is not in the document]
Comment (CIPop) [reply sdk/docs/core/README.md:232]: I'll change all to Paho since that's what we are prioritizing.

File: sdk/docs/core/README.md [hunk 3]
```diff
@@ -177,6 +177,138 @@ Note however that cancellation is performed as a best effort; it is not guarante
...
+
+The SDK is designed to avoid memory copy which results in the following ownership and life-time allocation requirements:
+
+- __Sending:__ The application may call APIs from any thread. The application is responsible with event-data allocation and initialization. The lifetime of the event-data associated with the API call will depend on the type of operation. It is safe to deallocate or re-use the event-data object after the operation completed (either successfully or with error). Please see the API documentation for more details. E.g. `az_mqtt5_rpc_client_begin_invoke_request` requires that the `az_mqtt5_rpc_client_invoke_req_event_data` object remains allocated until a `AZ_MQTT5_EVENT_RPC_CLIENT_RSP` is received.
+
+- __Receiving:__ When receiving events, the event-data object lifetime is guaranteed only for the duration of the callback. E.g. Receiving the `AZ_MQTT5_EVENT_RPC_CLIENT_RSP` event guarantees that the event-data is of `az_mqtt5_rpc_client_rsp_event_data` type. The lifetime
```
Comment (vaavva) [inline sdk/docs/core/README.md:282]: I think the end of this sentence got cut off

File: sdk/docs/core/README.md [hunk 4]
```diff
@@ -177,6 +177,138 @@ Note however that cancellation is performed as a best effort; it is not guarante
...
+
+- __Receiving:__ When receiving events, the event-data object lifetime is guaranteed only for the duration of the callback. E.g. Receiving the `AZ_MQTT5_EVENT_RPC_CLIENT_RSP` event guarantees that the event-data is of `az_mqtt5_rpc_client_rsp_event_data` type. The lifetime 
+
+Furthermore, long running operations within the callback will stall the network stack (this is intentional and will trigger TCP flow-control to allow the resource-constrained device more time to process inbound data). It is important to remember:
+
+   1. Never save event-data components such as `az_span`s without copying the referenced data into a memory location owned by the application.
+   1. Never block the callback with blocking I/O, synchronization constructs (mutex, semaphores, etc.) or long CPU-bound operations.
```
Comment (vaavva) [inline sdk/docs/core/README.md:287]: We currently have a mutex in the server sample mqtt_callback to ensure safe copying of the pending command data

File: sdk/docs/core/README.md [hunk 5]
```diff
@@ -177,6 +177,138 @@ Note however that cancellation is performed as a best effort; it is not guarante
...
+
+#### Credentials
+
+Credentials are handled by the `az_mqtt_connection_options` structure. If connection management is handled by the SDK, multiple X.509 client certificates can be used to allow for credential rotation.
+
+#### Implementation details
+
+For internal implementation details, please see the following documents:
+
+1. Each `az_mqtt_connection` object is an instance an [Asyncrhonous Pipeline](https://github.com/Azure/azure-sdk-for-c/blob/feature/v2/sdk/docs/core/az_event_pipeline.md) with each service API component being a member of the Policy Collection (orthogonal region).
```
Comment (vaavva) [inline sdk/docs/core/README.md:306]: Asyncrhonous Pipeline -> Asynchronous Pipeline

**Pull request merged**